src/hotspot/share/opto/escape.cpp

```diff
@@ -4712,13 +4712,21 @@ void ConnectionGraph::split_unique_types(GrowableArray<Node *>  &alloc_worklist,
       if (n == nullptr) {
         continue;
       }
+    } else if (n->Opcode() == Op_StrInflatedCopy) {
+      // Check direct uses of StrInflatedCopy.
+      // It is memory type Node - no special SCMemProj node.
     } else if (n->Opcode() == Op_StrCompressedCopy ||
                n->Opcode() == Op_EncodeISOArray) {
       // get the memory projection
       n = n->find_out_with(Op_SCMemProj);
       assert(n != nullptr && n->Opcode() == Op_SCMemProj, "memory projection required");
     } else {
+#ifdef ASSERT
+      if (!n->is_Mem()) {
```

Comment:
Would it make sense to turn this into a product check? Could we bail-out gracefully at this point?

Reply:
For first question - may be but we hit it first time since JDK 9. I think we should balance what is really needs to be checked in product and what can be done in debug VM. For this case I think checking in debug is fine.
For second question - not from EA but we can from compilation. We are in `split_unique_types()` where we start modifying graph already.